_rules/SC4-1-2-aria-attribute-has-valid-value.md

```diff
@@ -0,0 +1,255 @@
+- [Overview of possible ´aria-*´ attribute value types](https://www.w3.org/TR/wai-aria/#propcharacteristic_value)
+- [Uniform Resource Identifier (URI): Generic Syntax (RFC 3986)](https://www.ietf.org/rfc/rfc3986.txt)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Element with valid true/false value
+
+ ```html
+<div role="textbox" aria-required="true"></div>
+```
+
+#### Passed example 2
+ 
+Element with valid true/false/undefined value
+ 
+```html
+<div role="button" aria-expanded="undefined"></div>
+```
+
+#### Passed example 3
+ 
+Element with valid tristate value
+ 
+```html
+<div role="button" aria-pressed="mixed"></div>
+```
+
+#### Passed example 4
+ 
+Element with valid ID reference value
+ 
+```html
+<div role="textbox" aria-errormessage="my-error"></div>
+```
+
+#### Passed example 5
+ 
+Element with valid ID reference list value
+ 
+```html
+<div role="combobox" aria-owns="my-textbox my-grid"></div>
+```
+
+#### Passed example 6
+ 
+Element with valid integer value
+ 
+```html
+<div role="gridcell" aria-rowindex="2">Fred</div>
+```
+
+#### Passed example 7
+ 
+Element with valid number value
+ 
+```html
+<div role="spinbutton" aria-valuemin="1.0" aria-valuemax="2.0" aria-valuenow="1.5"></div>
+```
+
+#### Passed example 8
+ 
+Element with valid string value
+ 
+```html
+<div role="searchbox" aria-placeholder="MM-DD-YYYY">MM-DD-YYYY</div>
+```
+
+#### Passed example 9
+ 
+Element with valid token value, attribute inappropriate for the role.
+ 
+```html
+<div role="button" aria-orientation="horizontal"></div>
+```
+
+#### Passed example 10
+ 
+Element with valid token list value
+ 
+```html
+<div role="dialog" aria-dropeffect="copy move"></div>
+```
+
+### Failed
+
+#### Failed example 1
+
+Element with invalid true/false value
+
+```html
+<div role="textbox" aria-required="undefined"></div>
+```
+
+#### Failed example 2
+
+Element with invalid true/false/undefined value
+
+```html
+<div role="button" aria-expanded="mixed"></div>
+```
+
+#### Failed example 3
+
+Element with invalid tristate value
+
+```html
+<div role="button" aria-pressed="horizontal"></div>
+```
+
+#### Failed example 4
+
+Element with invalid ID reference value, space not allowed in a single ID
+
+```html
+<div role="textbox" aria-errormessage="error1 error2"></div>
+```
+
+#### Failed example 5
+
+Element with invalid integer value
+
+```html
+<div role="gridcell" aria-rowindex="2.5">Fred</div>
+```
+
+#### Failed example 6
+
+Element with invalid number value
+
+```html
+<div role="spinbutton" aria-valuemin="one" aria-valuemax="three" aria-valuenow="two"></div>
+```
+
+#### Failed example 7
+
+Element with invalid token value
+
+```html
+<div role="main" aria-live="nope"></div>
+```
+
+#### Failed example 8
+
+Element with invalid token list value
+
+```html
+<div role="dialog" aria-dropeffect="invalid move"></div>
+```
+
+#### Failed example 9
+
+Element with invalid empty aria attribute for token value
```

Comment:
I don't understand this one. It doesn't seem to match with the code provided below.

Reply:
@WilcoFiers, I think this example came from your original rule. 
Do you know what was the intention here?  
Or else I will just remove it, since I can't either find out now what it is supposed to illustrate that is not already covered by other test cases...